Packs/Citrix/Integrations/CitrixCloudEventCollector/CitrixCloudEventCollector.py

```diff
@@ -0,0 +1,205 @@
+        self.base_url = base_url
+        self.customer_id = customer_id
+        self.client_id = client_id
+        self.client_secret = client_secret
+        super().__init__(base_url=base_url, proxy=proxy, verify=verify)
+
+    def request_access_token(self):
+        demisto.debug("prepare to create access token")
+
+        headers = {"Accept": "application/json", "Content-Type": "application/x-www-form-urlencoded"}
+
+        data = {"grant_type": "client_credentials", "client_id": self.client_id, "client_secret": self.client_secret}
+
+        token_res = self._http_request(
+            "post", url_suffix=f"/cctrustoauth2/{self.customer_id}/tokens/clients", headers=headers, data=data
+        )
+
+        access_token = token_res.get("access_token")
+        if not access_token:
+            raise DemistoException("Failed to obtain access token from Citrix Cloud response.")
+
+        demisto.setIntegrationContext({ACCESS_TOKEN_CONST: access_token})
+        demisto.debug("access token created")
+        return access_token
+
+    def get_records(
+        self, start_date_time: str | None, end_date_time: str | None, continuation_token: str = None, limit: int = None
+    ):
+        # get access token value
+        integration_context = demisto.getIntegrationContext()
+        access_token = integration_context.get(ACCESS_TOKEN_CONST)
+
+        if not access_token:
+            access_token = self.request_access_token()
+
+        params = assign_params(
+            Limit=RECORDS_REQUEST_LIMIT,
+            continuationToken=continuation_token,
+            startDateTime=start_date_time,
+            endDateTime=end_date_time,
+        )
+
+        if limit and limit <= 200:
+            params["Limit"] = limit
+
+        headers = {
+            "Authorization": f"CwsAuth Bearer={access_token}",
+            "accept": "application/json",
+            "Citrix-CustomerId": self.customer_id,
+        }
+
+        demisto.info(f"Sending http request to get records with {params=}")
+        response = self._http_request(
+            "get", url_suffix="systemlog/records", headers=headers, params=params, ok_codes=[200, 401], resp_type="response"
+        )
+
+        if response.status_code == 401:
+            demisto.info("Access token expired; refreshing...")
+            access_token = self.request_access_token()
+            headers["Authorization"] = f"CwsAuth Bearer={access_token}"
+            demisto.info(f"Sending http request to get records with {params=}")
+
+            return self._http_request("get", url_suffix="systemlog/records", headers=headers, params=params)
+        else:
+            return response.json()
+
+    def get_records_with_pagination(self, limit: int, start_date_time: str | None, end_date_time: str | None = None):
+        records: list[dict] = []
+        continuation_token = None
+        raw_res = None
+
+        while len(records) < int(limit):
```

Comment:
The pagination logic doesn't handle potential duplicate events when using timestamp-based pagination. If multiple events have the same timestamp as the last fetched event, they could be duplicated across fetch cycles.